pkg/types/conversion/tfjson/tfjson.go

```diff
@@ -0,0 +1,221 @@
+/*
+ Copyright 2022 The Crossplane Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package tfjson
+
+import (
+	tfjson "github.com/hashicorp/terraform-json"
+	schemav2 "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+	"github.com/zclconf/go-cty/cty"
+)
+
+// GetV2ResourceMap converts input resource schemas with
+// "terraform-json" representation to terraform-plugin-sdk representation which
+// is what Terrajet expects today.
+//
+// What we are trying to achieve here is to convert a lower level
+// representation of resource schema map, e.g. output of `terraform providers schema -json`
+// to plugin sdk representation. This is mostly the opposite of what the
+// following method is doing: https://github.com/hashicorp/terraform-plugin-sdk/blob/7e0a333644f1971a936995677b7a106140a0659f/helper/schema/core_schema.go#L43
```

Comment:
I'm not sure how long we'd like to keep the backward compatibility, especially if there is no CRD API change, but the next step could be to make builder work with CLI schema directly and do the `sdk -> cli schema` conversion using this referenced tool. You may consider doing that in this PR, too, if the conversions we have here are too similar to that file anyway.

Reply:
As mentioned in the next comment lines, I agree that ultimately our builder should work with cli schema. However, I still see value to experiment current approach with new providers for a while.

I liked the idea of working with CLI schema in type builder but doing the conversion from `sdk -> cli schema`, but I wouldn't prefer to do in this PR because:

- Making type builder working with CLI would require a good amount of work and I would not prefer to delay this feature for new providers since there are some people struggling with the current approach, i.e. importing tf provider go package.
- Since the `cli schema` is a more limited representation than the `sdk` one, we would still lose some information, e.g. int/float distinction or lack of required attribute for a deeply nested object field, etc. So, at this point (having one conversion already implemented), I don't see much benefit of spending time the other way. 

Stating this, I believe we can prioritize to work on a follow-up PR updating type builder to work with cli schema and speed up the experimenting process.